avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroSerDe.java

```diff
@@ -86,14 +90,26 @@ protected void configureClientProperties(AbstractKafkaAvroSerDeConfig config) {
   /**
    * Get the subject name for the given topic and value type.
    */
-  protected String getSubjectName(String topic, boolean isKey, Object value) {
-    if (isKey) {
-      return keySubjectNameStrategy.getSubjectName(topic, isKey, value);
+  protected String getSubjectName(String topic, boolean isKey, Object value, Schema schema) {
+    Object subjectNameStrategy = subjectNameStrategy(isKey);
+    if (subjectNameStrategy instanceof SubjectNameStrategy) {
+      return ((SubjectNameStrategy) subjectNameStrategy).getSubjectName(topic, isKey, schema);
     } else {
-      return valueSubjectNameStrategy.getSubjectName(topic, isKey, value);
+      return ((io.confluent.kafka.serializers.subject.SubjectNameStrategy) subjectNameStrategy)
+          .getSubjectName(topic, isKey, value);
```

Comment:
I asked whether our strategies should still implement the older interface for backward compatibility reasons. If so, then our implementations will implement both interfaces, so this check should perform the instanceof with the newer interface *first* so that the more efficient interface/implementation is chosen.